doofinder-for-woocommerce/includes/class-add-to-cart.php

```diff
@@ -41,23 +41,16 @@ public function __construct()
 
         add_action('wp_ajax_doofinder_ajax_add_to_cart',  array(__CLASS__, 'doofinder_ajax_add_to_cart'));
         add_action('wp_ajax_nopriv_doofinder_ajax_add_to_cart',  array(__CLASS__, 'doofinder_ajax_add_to_cart'));
-
-        // Register custom WP REST Api endpoint
-        add_action('rest_api_init', function () {
-            register_rest_route('doofinder-for-wc/v1', '/product-info/(?P<id>\d+)', array(
-                'methods' => ['GET'],
-                'callback' => array(__CLASS__, 'product_info'),
-                'permission_callback' => '__return_true'
-            ));
-        });
+        add_action('wp_ajax_doofinder_get_product_info',  array(__CLASS__, 'product_info'));
+        add_action('wp_ajax_nopriv_doofinder_get_product_info',  array(__CLASS__, 'product_info'));
     }
 
     /**
-     * Callback for WP Rest Api item-info endpoint
+     * Returns the product info for a given id
      */
-    public static function product_info(\WP_REST_Request $request)
+    public static function product_info()
     {
-        $post_id = $request->get_param("id");
+        $post_id = $_REQUEST['id'] ?? NULL;
```

Comment:
Is this working? JS is case sensitive and I don't think it is gonna understand null in uppercase

Reply:
This is not a problem as it is php. 
What I should do is to return false when the product id is missing